providers/google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/strategy/CreateNodesWithGroupEncodedIntoNameThenAddToSet.java

```diff
@@ -147,18 +149,27 @@
    private void configureNetworking(String group, GoogleComputeEngineTemplateOptions options, Location location) {
       String networkName = null;
       Network network = null;
+      String net = null;
+      boolean isFullURI = false;
 
       if (options.getNetworks().isEmpty()) {
          networkName = DEFAULT_NETWORK_NAME;
       } else {
          Iterator<String> iterator = options.getNetworks().iterator();
-         networkName = nameFromNetworkString(iterator.next());
+         net = iterator.next();
+         if (net.startsWith("projects")) {
+            isFullURI = true;
+         }
+         networkName = nameFromNetworkString(net);
          checkArgument(!iterator.hasNext(),
                "Error: Please specify only one network/subnetwork in TemplateOptions when using GCE.");
       }
 
       String region = ZONE == location.getScope() ? location.getParent().getId() : location.getId();
-      Optional<Subnetwork> subnet = subnetworksMap.getUnchecked(fromRegionAndName(region, networkName));
+      Optional<Subnetwork> subnet;
+
+      subnet = isFullURI ? Optional.fromNullable(resources.subnetwork(URI.create(net))) : subnetworksMap.getUnchecked(fromRegionAndName(region, networkName));
```

Comment:
Could this logic be abstracted in the subnet cache? Otherwise, we are catching subnets only if the provided ID is not the full URI, which does not seem right?
This way you save an extra call in the adapter. With the current code, you are calling the `resources` API here and calling it again in the adapter.

Reply:
Ok. But a shared VPC is identified by his `host project`, `region` and `name` and a classical subnetwork is only identified by his `region` and `name`, the `project` is given by the service account. 
In my opinon, if a user wants to use a shared VPC, it'll have to give the full URI and if he wants to use a classical subnetwork, it'll have to give only the `region` and `name`. In this way, we are not breaking the compatibility. 
This is why, I am checking if the given `net` is like `projects/<project>/regions/<region>/subnetworks/<subnetwork>`.

Reply:
Ok. The main issue here is that the cache key is `RegionAndName`. What about creating a different class to be used with these loaders that can cache shared resources, say `RegionAndNameInProject`? In the case of shared resources we can build it from the full URI, and for others we can just get injected the current project (we already have the project bound to the Guice contecxt as `@Named(PROJECT_NAME) String currentProject`) and build it too. This way we can still have backwards compatibility and properly cache all subnets whether they are shared or not, saving unnecessary repeated API calls. WDYT?